### PR TITLE
added typings and docs for vectorized mobjects and bezier related functions

### PR DESCRIPTION
**Pull request by huguesdevimeux**
## Motivation
This part of the codebase is quite hard to understand and a little bit unordered in some PR. 
This PR aims to add sustainable docs and comments to make comprehension easier and to prepare some eventual refactors.  

## Testing Status
Tests pass. 

## Further Comments

This is is only a proposal. _Please_ tell me any english of comprehension mistakes!  

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Oneline Summary of Changes

```
- Added typings and docs for vectorized mobjects and bezier related functions  (:pr:`961`)
```

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

File: manim/mobject/types/vectorized_mobject.py [hunk 1]
```diff
@@ -433,11 +435,28 @@ def set_points(self, points):
...
+        """Given two sets of anchors and handles, process them to set them as anchors and handles of the VMobject. anchors1[i], handles1[i], handles2[i] and anchors2[i] define the
+        i-th bezier curve of the vmobject.
+
+        There are four hardcoded paramaters and this is a problem as it makes the number of points per cubic curve unchangeable from 4. (two anchors and two handles).
+
+        Returns
+        -------
+        VMobject
+            for chaining.
+        """
```
Comment (leotrs) [inline manim/mobject/types/vectorized_mobject.py:454]: Nitpick: review https://numpydoc.readthedocs.io/en/latest/format.html#sections, specifically the part about the "short summary": "A one-line summary that does not use variable names or the function name". If the docstring needs to refer to variable names, it is recommended to do that in the notes section or in the extended summary part.

File: manim/mobject/geometry.py [hunk 2]
```diff
@@ -919,6 +919,7 @@ def __init__(self, points, **kwargs):
 class Polygon(VMobject):
     def __init__(self, *vertices, color=BLUE, **kwargs):
         VMobject.__init__(self, color=color, **kwargs)
+        # There are actually four corner, and the first one is repeated twice to form the four vertices.
```
Comment (leotrs) [inline manim/mobject/geometry.py:922]:
```suggestion
        # There are actually four corners, and the first one is repeated twice to form the four vertices.
```

**Comment by leotrs** [comment]
Didn't approve bc it's not clear whether this is WIP or completely done?

File: manim/mobject/types/vectorized_mobject.py (hunk 1 above)
Comment (huguesdevimeux) [reply manim/mobject/types/vectorized_mobject.py:454]: I didn't know that. thanks !

**Comment by huguesdevimeux** [comment]
@leotrs It's done IMO

File: manim/mobject/types/vectorized_mobject.py [hunk 3]
```diff
@@ -458,13 +477,27 @@ def start_new_path(self, point):
         self.append_points([point])
         return self
 
-    def add_cubic_bezier_curve(self, anchor1, handle1, handle2, anchor2):
+    def add_cubic_bezier_curve(
+        self, anchor1: np.ndarray, handle1: np.ndarray, handle2: np.ndarray, anchor2
```
Comment (kolibril13) [inline manim/mobject/types/vectorized_mobject.py:481]: can you add an example here?
e.g. this one 
```py
class BezierSpline(Scene):
    def construct(self):
        np.random.seed(42)
        area = 4
        p1 = np.array([-3, 1, 0])
        p1b = p1 + [1, 0, 0]
        d1 = Dot(point=p1).set_color(BLUE)
        l1 = Line(p1, p1b)
        self.add(l1, d1)
        p2 = np.array([3, -1, 0])
        p2b = p2 - [1, 0, 0]
        d2 = Dot(point=p2).set_color(RED)
        l2 = Line(p2, p2b)
        self.add(l2, d2)
        l1.add_cubic_bezier_curve(p1b, p1b + 3 * RIGHT, p2b - 3 * RIGHT, p2b)
```
image

Comment (huguesdevimeux) [reply manim/mobject/types/vectorized_mobject.py:481]: Hmmmm do you think this is meant to be used by the end-user?

Comment (kolibril13) [reply manim/mobject/types/vectorized_mobject.py:480]: good question.
I think the possibility to make a bezier spline would be a nice and useful feature. But I am not sure if it should be used like this.

Comment (huguesdevimeux) [reply manim/mobject/types/vectorized_mobject.py:480]: Yes, I'm not sure too. For me, this is mean to be used for an internal use only.

Comment (kolibril13) [reply manim/mobject/types/vectorized_mobject.py:480]: okay, then we leave it without an example

**Review by leotrs — APPROVED** [review]

File: manim/mobject/types/vectorized_mobject.py (hunk 3 above)
Comment (huguesdevimeux) [reply manim/mobject/types/vectorized_mobject.py:480]: Additonnaly, there is already `CubicBezier` for that IIRC

**Pull request merged**